### PR TITLE
Audit `ChatMessageLayoutOptionsResolver`

### PR DESCRIPTION
**Pull request by evsaev**
**This PR**:
- exposes `isMessageLastInSequence` on `ChatMessageLayoutOptionsResolver`
- adds missing tests to `ChatMessageLayoutOptionsResolver` type

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+                    && !channel.isDirectMessageChannel
+            }
+        }
+    }
+
+    func test_quotedMessage() {
+        // Iterate channel history and check `.quotedMessage` is there if conditions are met
+        assertOptionIsIncluded(.quotedMessage) { message, _ in
+            message.quotedMessage != nil && message.deletedAt == nil
+        }
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift:113]: I don't think this is the best way to test this. Tests should be as dummy as possible and should not contain any logic, and with this, we are not really testing that "quotedMessage" added the `.quotedMessage` option. For example, if there was a bug, and `isSentByCurrentUser` option

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+            )
+
+            // Assert the `option` is there or not depending on the requirement
+            XCTAssertEqual(
+                layoutOptions.contains(option),
+                mustBeIncludedFor(message, isStandaloneOrLastInSequence)
+            )
+        }
+    }
+}
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift:194]: Like I said above, with this helper, is not possible to check the exact message which added the option, which will not catch possible future bugs

File: Sources/StreamChatUI/Utils/ChatMessage+Extensions.swift [hunk 3]
```diff
@@ -28,10 +28,7 @@ extension _ChatMessage {
 
     /// A boolean value that checks if the message is the root of a thread.
     var isRootOfThread: Bool {
-        let isThreadStart = replyCount > 0
-        let isThreadReplyInChannel = showReplyInChannel
-
-        return isThreadStart || isThreadReplyInChannel
+        replyCount > 0
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/Utils/ChatMessage+Extensions.swift:31]: 👍

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift (hunk 1 above)
Comment (evsaev) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift:113]: Yeah, I see what you mean. Makes sense 👍 Let me have one more iteration on this

Comment (evsaev) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift]: ✅ Updated

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift (hunk 2 above)
Comment (evsaev) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver_Tests.swift]: ✅ Removed

**Review by nuno-vieira — APPROVED** [review]
LGTM

**Pull request merged**